src/pl/senla/hotel/repository/RepositoryFacilityCollection.java

```diff
@@ -1,23 +1,26 @@
 package pl.senla.hotel.repository;
 
+import pl.senla.hotel.annotations.di.AppComponent;
+import pl.senla.hotel.annotations.di.GetInstance;
 import pl.senla.hotel.entity.facilities.HotelFacility;
 import pl.senla.hotel.storage.DataStorage;
-import pl.senla.hotel.storage.DataStorageFacility;
 
 import java.util.List;
 
+@AppComponent
 public class RepositoryFacilityCollection implements Repository<HotelFacility> {
 
     private static Repository<HotelFacility> repositoryFacility;
+    @GetInstance(beanName = "DataStorageFacility")
     private final DataStorage<HotelFacility> priceList;
 
-    private RepositoryFacilityCollection() {
-        this.priceList = DataStorageFacility.getDataStorageFacility();
+    private RepositoryFacilityCollection(DataStorage<HotelFacility> priceList) {
+        this.priceList = priceList;
     }
 
-    public static Repository<HotelFacility> getRepositoryFacility(){
+    public static Repository<HotelFacility> getSingletonInstance(DataStorage<HotelFacility> priceList){
```

Comment:
get rid of all singletons in your application, you should have a special context where you keep all created instances and should manage them from this context

Reply:
Thank you. Excellent idea.
I have broken my mind trying to find way how get instances from private constructors or method "getSingletonInstance" with huge amount of other parametrs-singlotones.